### DISCUSSION TITLE
Adding 'batch mode' to MessagePassing

### DISCUSSION THREAD

**Discussion by cclough** [danielegrattarola/spektral#361] [Q&A] [answered]
Thank you for this excellent project.

I'd like to use BatchLoader with a model using MessagePassing.  I'm trying to decide the best way to approach the problem:
1. Overriding methods in MessagePassing
2. Building up from Conv/GCNConv, as recommended at the end of https://graphneural.network/creating-layer/

Any advice on how best to approach this?

For 1. could there be a quick fix to 'get_inputs', or do you think there are many changes needed?

**Comment by danielegrattarola** [comment, marked as answer]
A few thoughts.

1. Batch mode is mostly there because it feels familiar to people coming to GNNs from typical ML, but in most cases you're better off using disjoint mode. So unless you need some specific pooling layer that only works in batch mode, I would suggest you to re-evaluate whether you really need batch mode (I don't know the specific of your project, I'm just giving my 2 cents)
2. The best way to go about it would be to subclass `Conv`. There is no advantage of using `MessagePassing` over `Conv` anyway, at the end you just need to implement your desired tensor manipulation inside of the `call` method. `MessagePassing` only gives you a bit of structure in defining the tensor manipulation, and also checks that the input is in the correct format. 
3. That said, if you want to use `MessagePassing` you just have to make `get_inputs` return the inputs, removing any checks. 

Cheers

**Comment by cclough** [comment]
@danielegrattarola sincere thanks for the quick and thorough response. Much appreciated. Thanks to your advice we figured out that we could do what we wanted with Disjoint mode.
